File: first10/Otoshidama.jl
using Printf

line = readline()
N, Y = split(line, " ")
N = parse(Int64, N)
Y = parse(Int64, Y)


function otoshi(N, Y)
    flag = false
    for ten = 0:N
        for fiv = 0:(N - ten)
            one = N - ten - fiv
            if ten*10000 + fiv*5000 + one*1000 == Y
                @printf "%d %d %d" ten fiv one
                flag = !flag
                return 0
            end
        end
    end

    if !flag
        println("-1 -1 -1")
    end
        
end

otoshi(N, Y)

File: first10/Otoshidama.py

N, Y = map(int, input().split())




def otoshi(N, Y):
    flag = False
    for ten in range(N+1):
        for fiv in range(N - ten +1):
            one = N - ten - fiv
            if ten*10000 + fiv*5000 + one*1000 == Y:
                print("{} {} {}".format(ten, fiv, one))
                flag = not flag
                return 0
            else:
                pass
    if not flag:
        print("-1 -1 -1")
        
otoshi(N, Y)